zookeeper-api/src/main/java/org/apache/helix/zookeeper/api/factory/RealmAwareZkClientFactory.java

```diff
@@ -31,25 +34,21 @@
    * Build a RealmAwareZkClient using specified connection config and client config.
    * @param connectionConfig
    * @param clientConfig
-   * @param metadataStoreRoutingData
    * @return HelixZkClient
```

Comment:
Document when/why IOExeption would be thrown and why InvalidRoutingDataException would be thrown

Reply:
That is correct. I will update the JavaDoc :)